src/Grid/grid.jl

```diff
@@ -765,8 +765,11 @@ for INDEX in (:VertexIndex, :EdgeIndex, :FaceIndex)
         #To be able to do a,b = faceidx
         Base.iterate(I::($INDEX), state::Int=1) = (state==3) ?  nothing : (I[state], state+1)
 
-        #For (cellid, faceidx) in faceset
-        Base.in(v::Tuple{Int, Int}, s::Set{$INDEX}) = in($INDEX(v), s)
+        #Necessary to check if, e.g. `(cellid, faceidx) in faceset`
+        Base.isequal(x::$INDEX, y::$INDEX) = x.idx[1] == y.idx[1] && x.idx[2] == y.idx[2]
+        Base.isequal(x::Tuple{Int, Int}, y::$INDEX) = x[1] == y.idx[1] && x[2] == y.idx[2]
+        Base.isequal(y::$INDEX, x::Tuple{Int, Int}) = x[1] == y.idx[1] && x[2] == y.idx[2]
+        Base.hash(x::$INDEX,h::UInt) = Base.hash(x.idx,h)
```

Comment:
Sorry, didn't spot this in the first round. 
```suggestion
        Base.hash(x::$INDEX, h::UInt) = hash(x.idx, h)
```
Not required to remove the `Base.` in the last, not sure if it was intentional to highlight that we use the function from `Base`?

Reply:
yeah, it was somewhat intentional to show that we forward to the `Base` function, but I can remove it!

Reply:
Fine to keep to IMO, makes it clear for the one reading the code!